src/Symfony/Component/Translation/LocaleSwitcher.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class LocaleSwitcher implements LocaleAwareInterface
+{
+    /**
+     * @param LocaleAwareInterface[] $localeAwareServices
+     */
+    public function __construct(
+        private string $locale,
+        private iterable $localeAwareServices,
+        private ?RequestContext $requestContext = null
```

Comment:
I'm not entirely happy this object needs to be aware of `RequestContext`. What about these ideas?

1. In FrameworkBundle add a `RequestContextAwareLocaleSwitcher` that extends this
2. Add a wrapper service for `RequestContext` that implements `LocaleAwareInterface` and inject as part of `$localeAwareServices`. This would be a bit more complicated and require a new compiler pass I believe (we couldn't tag this decorator as `kernel.locale_aware`).

Reply:
2 sounds nice :)

Reply:
Went with 2.